docs/modules/ROOT/pages/guides/how-to-dynamic-client-registration.adoc

```diff
@@ -0,0 +1,174 @@
+[[how-to-dynamic-client-registration]]
+= How-to: Register a client dynamically
+:index-link: ../how-to.html
+:docs-dir: ..
+
+This guide shows how to configure OpenID Connect Dynamic Client Registration 1.0 in Spring Authorization Server and
+walks through an example of how to register a client. Spring Authorization Server implements https://openid.net/specs/openid-connect-registration-1_0.html[OpenID Connect Dynamic Client Registration 1.0]
+specification, gaining the ability to dynamically register and retrieve OpenID clients.
+
+- xref:guides/how-to-dynamic-client-registration.adoc#enable[Enable Dynamic Client Registration in Spring Authorization Server]
+- xref:guides/how-to-dynamic-client-registration.adoc#configure-initial-client[Configure initial client]
+- xref:guides/how-to-dynamic-client-registration.adoc#fetch-initial-access-token[Fetch initial access token]
+- xref:guides/how-to-dynamic-client-registration.adoc#register-client[Register a client]
+- xref:guides/how-to-dynamic-client-registration.adoc#fetch-client[Fetch client]
+
+[[enable]]
+== Enable Dynamic Client Registration in Spring Authorization Server
+
+By default, dynamic client registration functionality is turned off in Spring Authorization Server.
+To enable, add the following configuration:
+
+[[sample.dcrAuthServerConfig]]
+[source,java]
+----
+include::{examples-dir}/main/java/sample/dcr/DcrConfiguration.java[]
```

Comment:
We should provide a fully working configuration. Please take a look at 

https://github.com/spring-projects/spring-authorization-server/blob/e5fcbe422be1a73240ac7d8b40f165651fb01ab8/docs/src/main/java/sample/extgrant/SecurityConfig.java#L67

and how it makes use of `@fold:on` and `@fold:off` to only show the relevant configuration - but it can be expanded to show the complete configuration.